### PR TITLE
Icicle Burst can now only be played when enemy units were frozen this turn

### PR DESCRIPTION
**Pull request by Neicigam**

### CODE REVIEW

File: src/components/DeckMechanisms/index.js [hunk 1]
```diff
@@ -366,6 +385,9 @@ export default class DeckMechanisms extends React.Component {
             : 0
         newState.specifics.liveDawnsparks = false
       }
+      if (newState.specifics.potentialFrozenEnemiesInAColumn) {
+        newState.specifics.potentialFrozenEnemiesInAColumn = 0
+      }
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:390]: You can skip the condition entirely here and simply reassign:

```js
newState.specifics.potentialFrozenEnemiesInAColumn = 0
```

File: src/components/DeckMechanisms/index.js [hunk 2]
```diff
@@ -125,14 +129,29 @@ export default class DeckMechanisms extends React.Component {
...
+            newState.specifics.potentialFrozenEnemiesInAColumn += 2
+            break
+          case 'W6':
+            // Moment's Peace
+            newState.specifics.potentialFrozenEnemiesInAColumn += 3
+            break
+          case 'W11':
+            // Midwinter Chaos
+            newState.specifics.potentialFrozenEnemiesInAColumn += 4
+            break
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:152]: I’m not sure how much value there is in counting enemies. Maybe we could simplify it like this:

```js
case 'W2':
case 'W6':
case 'W11':
  newState.specifics.potentiallyFrozenUnits = true
  break
```

Comment (Neicigam) [reply src/components/DeckMechanisms/index.js:153]: I was planning to implement Zhevana's ability after - that's why I'm only counting in a column

File: src/components/DeckMechanisms/index.js (hunk 1 above)
Comment (Neicigam) [reply src/components/DeckMechanisms/index.js:401]: I wanted to keep it consistent with the Frozen Core test which reassigns mana only when it's not equal to 0, even though the test could be removed. It's probably better to remove both though

File: src/components/DeckMechanisms/index.js [hunk 3]
```diff
@@ -402,10 +416,14 @@ export default class DeckMechanisms extends React.Component {
...
-    const unplayableSpells = ['W1', 'S10', 'N15']
+
+    if (id === 'W1' && !this.state.specifics.potentialFrozenEnemies) {
+      return false
+    }
 
     if (this.state.turn === 1) {
-      if (this.state.noUnitsOnFirstTurn) {
+      const unplayableSpells = ['W1', 'S10', 'N15']
+      if (this.state.specifics.noUnitsOnFirstTurn) {
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:426]:
```suggestion
      const unplayableSpells = ['W1', 'S10', 'N15']
      
      if (this.state.specifics.noUnitsOnFirstTurn) {
```

File: src/components/DeckMechanisms/index.js [hunk 4]
```diff
@@ -20,8 +20,12 @@ export default class DeckMechanisms extends React.Component {
       hand: [],
       RNG: 'REGULAR',
       hasCycledThisTurn: false,
-      noUnitsOnFirstTurn: true,
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:23]: Good thinking to move it in `specifics`. Way better. :)

**Pull request merged**